sentry-android-core/src/main/java/io/sentry/android/core/ViewHierarchyEventProcessor.java

```diff
@@ -108,12 +117,28 @@ public static ViewHierarchy snapshotViewHierarchy(
     }
 
     try {
-      final @NotNull ViewHierarchy viewHierarchy = snapshotViewHierarchy(decorView);
-      return viewHierarchy;
+      if (mainThreadChecker.isMainThread()) {
+        return snapshotViewHierarchy(decorView);
+      } else {
+        final CountDownLatch latch = new CountDownLatch(1);
+        final AtomicReference<ViewHierarchy> viewHierarchy = new AtomicReference<>(null);
+        activity.runOnUiThread(
```

Comment:
`l`: potentially we could do `handler.postAtFrontOfQueue` instead to capture the VH/SS as soon as possible, but I don't know what could be the potential side effects of this. WDYT?

Reply:
Hmm that's a good one. [Reading up on the docs](https://developer.android.com/reference/android/os/Handler#postAtFrontOfQueue(java.lang.Runnable)), I wouldn't do it - at least not within this PR - maybe as a follow up investigation, as it could be relevant for other parts within our code as well?

Reply:
Avoid `activity.runOnUiThread` in favor of `MainLooperHandler` for testability.
See the previous comment related to this issue as well:
https://github.com/getsentry/sentry-java/pull/2281#discussion_r990935530
Added a new comment here https://github.com/getsentry/sentry-java/issues/2515#issuecomment-1556754682